packages/components/src/tools-panel/tools-panel-item/hook.ts

```diff
@@ -11,8 +11,9 @@ import * as styles from '../styles';
 import { useToolsPanelContext } from '../context';
 import { useContextSystem } from '../../ui/context';
 import { useCx } from '../../utils/hooks/use-cx';
+import type { ToolsPanelItemProps } from '../types';
 
-export function useToolsPanelItem( props ) {
+export function useToolsPanelItem( props: ToolsPanelItemProps ) {
```

Comment:
Writing a note for myself really, which I hope will be answered during review 😄 

Does this need to hook into [PolymorphicComponentProps](https://github.com/WordPress/gutenberg/blob/2b974ceda8070ee9460f8457dbbd29db4a2abfa8/packages/components/src/ui/context/polymorphic-component.ts)?

E.g., 

```
export function useToolsPanel( props: PolymorphicComponentProps<ToolsPanelProps, 'div'> ) {
```

Just looking at the other hooks and the [recommendations on the original PR](https://github.com/WordPress/gutenberg/pull/32392#discussion_r679139862).

Reply:
Good question, will take a look at that.

Reply:
Have added in the `PolymorphicComponentProps` now.